torch/utils/data/graph_settings.py

```diff
@@ -31,5 +31,5 @@ def apply_shuffle_settings(datapipe, shuffle):
         graph = torch.utils.data.graph.traverse(datapipe, only_datapipe=True)
         all_pipes = get_all_graph_pipes(graph)
         for pipe in all_pipes:
-            if hasattr(pipe, 'set_shuffle_settings'):
-                pipe.set_shuffle_settings(shuffle)
+            if hasattr(pipe, 'set_shuffle'):
```

Comment:
Shouldn't we do `isinstance(pipe, Shuffler)` here? Checking for a public attribute seems sketchy.